packages/edit-site/src/components/sidebar/global-styles-v2/screens/typography-element.js

```diff
@@ -0,0 +1,262 @@
+		defaultValue: null,
+	},
+];
+
+const TypographyOptions = () => {
+	const {
+		attributes: { styles },
+		toggleAttribute,
+	} = useStyleAttributesForScreen();
+
+	const options = typographyOptions.map( ( o ) => {
+		const isActive = ! isUndefined( styles[ o.value ] );
+		return {
+			...o,
+			isActive,
+			onClick: () => toggleAttribute( o.value, o.defaultValue ),
+			icon: isActive ? check : undefined,
+		};
+	} );
+
+	return <DropdownMenu controls={ options } />;
+};
+
+const Header = ( { title } ) => {
+	return (
+		<ScreenHeader
+			action={ <TypographyOptions /> }
+			back="/typography"
+			description={ `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla
+				congue finibus ante vel maximus.` }
+			title={ title }
+		/>
+	);
+};
+
+function useStyleAttributesForScreen() {
+	const { id } = useNavigatorParams();
+	const appState = useAppState();
+	const { get, set } = appState;
+
+	let index;
+	const attributes = appState.typography.elements.find( ( e, i ) => {
+		if ( e.slug === id ) {
+			index = i;
+			return true;
+		}
+		return false;
+	} );
+
+	const getPath = ( attr ) =>
+		`typography.elements[${ index }].styles.${ attr }`;
+
+	const getAttribute = ( attr ) => attributes.styles[ attr ];
+	const setAttribute = ( attr ) => ( next ) => {
+		const path = getPath( attr );
+		const value = isUndefined( next ) ? null : next;
+		set( path, value );
+	};
+	const toggleAttribute = ( attr, defaultValue = undefined ) => {
+		const path = getPath( attr );
+		const current = get( path );
+		if ( isUndefined( current ) ) {
+			set( path, defaultValue );
+		} else {
+			set( path, undefined );
+		}
+	};
+
+	const getIsDefined = ( attr ) => {
+		const path = getPath( attr );
+		const current = get( path );
+		return ! isUndefined( current );
+	};
+
+	return {
+		getAttribute,
+		setAttribute,
+		toggleAttribute,
+		getIsDefined,
+		attributes,
+	};
+}
+
+const fontWeights = [ 100, 200, 300, 400, 500, 600, 700, 800, 900 ];
+const fontWeightOptions = fontWeights.map( ( value ) => ( {
+	value,
+	label: value,
+} ) );
+
+const RenderComponent = ( { children, prop } ) => {
+	if ( isUndefined( prop ) || prop === false ) return null;
+	return children;
+};
+
+export const TypographyElementScreen = () => {
+	const { id } = useNavigatorParams();
+	const title = startCase( id );
+	const {
+		getAttribute,
+		getIsDefined,
+		setAttribute,
+	} = useStyleAttributesForScreen();
+	const bindAttribute = ( attr, props = {} ) => ( {
+		...props,
+		value: getAttribute( attr ) || props.defaultValue,
+		onChange: setAttribute( attr ),
+	} );
+
+	return (
+		<Screen>
+			<CardBody>
+				<VStack spacing={ 8 }>
+					<Header title={ title } />
+					<Panel>
+						<RenderComponent prop={ getIsDefined( 'fontFamily' ) }>
+							<TextControl
+								label="Font"
+								{ ...bindAttribute( 'fontFamily' ) }
+							/>
+						</RenderComponent>
+						<RenderComponent prop={ getIsDefined( 'fontSize' ) }>
+							<Grid>
+								<FontSizePicker
```

Comment:
`FontSizePicker` needs some love in this context. We need to compare it to the g2 implementation and find a better way to have the slider inline with the number field for font size picker.